sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/AuthorizationCodeCredentialBuilder.java

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.identity.implementation.util.ValidationUtil;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * Fluent credential builder for instantiating a {@link AuthorizationCodeCredential}.
+ *
+ * @see AuthorizationCodeCredential
+ */
+public class AuthorizationCodeCredentialBuilder extends AadCredentialBuilderBase<AuthorizationCodeCredentialBuilder> {
+    private static final ClientLogger LOGGER = new ClientLogger(AuthorizationCodeCredentialBuilder.class);
+
+    private String authCode;
+    private String redirectUri;
+
+    /**
+     * Sets the authorization code on the builder. This is required for building an
+     * {@link AuthorizationCodeCredential}, but is not required for constructing the
+     * login URL.
+     *
+     * @param authCode the authorization code acquired from user login
+     * @return the AuthorizationCodeCredentialBuilder itself
+     */
+    public AuthorizationCodeCredentialBuilder authorizationCode(String authCode) {
+        this.authCode = authCode;
+        return this;
+    }
+
+    /**
+     * Sets redirect URL for the Oauth 2.0 login request, which must be
+     * registered as a valid reply URL on the application. the authorization code
+     * will be sent to this URL so it must be listening on this server and is able
+     * to complete the {@link AuthorizationCodeCredential} construction from there.
+     *
+     * @param redirectUri the redirect URI to send the authorization code
+     * @return the AuthorizationCodeCredentialBuilder itself
+     */
+    public AuthorizationCodeCredentialBuilder redirectUri(String redirectUri) {
+        this.redirectUri = redirectUri;
+        return this;
+    }
```

Comment:
This method JavaDoc mentions "URL" a few times, but the API itself mentions "URI' a few times. Can you standardise on one?

Reply:
Changed everything to redirect URI but also added note this is also called the "Reply URL" in some contexts.